Update rubocop.yml template

fastjames opened the pull request:
The template contains cops that are no longer supported.

fastjames committed:
Update rubocop.yml templates

The rubocop.yml for this repo, as well as the template rubocop.yml,
contain outdated cops. We update both files to the latest cop names, and
set them to inherit from the relaxed ruby style guide.

tvdeyen reviewed (approved):
Thanks. Since this is 6 months old, its worth checking if other cops need to be updated as wel. Would you mind to check again?

fastjames commented:
@tvdeyen good point! I fixed several outdated cop names, and also switched to inheriting from the relaxed ruby style guide. I also fixed the resulting cop violations, since they were minor misdemeanors. :policeman:  I plan to squash the commits before merge.

tvdeyen commented on .rubocop-https---relaxed-ruby-style-rubocop-yml:
```diff
@@ -0,0 +1,166 @@
+# Relaxed.Ruby.Style
```
I think this file should live under 

`lib/solidus_cmd/templates/extension`

no?

tvdeyen reviewed (changes requested):
Thanks. I guess the updated rubocop file needs to be moved into the extension template folder. Also, please leave the rubocop violation fixes in a separate commit, for better git history

fastjames committed:
Fix rubocop violations in repo

Updating to the most recent relaxed ruby style guide would produce some
cop violations, so we want to fix those before the switch.

fastjames commented:
@tvdeyen thanks for the excellent feedback, you were correct in pointing out that I only updated one of the two rubocop yml files in the repo. While I fixed up the one for this repo, I skipped over the template rubocop.yml that gets copied into new extensions. I have made those changes as well, and rebased the PR into 2 commits: a pre-emptive commit fixing the cop violations, and a commit containing the yml changes.

One thing I left out is the cached files from relaxed.ruby.style (you had pointed out that file in an earlier comment). I am wondering if we should commit that file or gitignore it. According to [the rubocop docs](https://github.com/rubocop-hq/rubocop/blob/master/manual/configuration.md#inheriting-configuration-from-a-remote-url) the file will be updated if the remote is different and it's >24 hours old. I'm OK with including the files, but I wanted to get a second opinion before I did so.

jarednorman commented:
I've opened #24 which accomplishes roughly the same thing as this one (sorry, didn't see this one first.) I think my approach of tracking the gem to ensure the relaxed ruby style and rubocop stay in sync will require less maintenance in the long run. I also chose not to address any of the violations, as I didn't think it was worth the noise, but I'm happy to fix them if that's preferred.

kennyadsl commented:
Closing after #24 merge. Thanks!
